### PR TITLE
Bug#80 Too many sync tasks on UI thread for MatchActivity

### PR DESCRIPTION
**Pull request by limphilip**
#80 

This pull request is to solve the synchronicity issue for the `MatchActivity `causing it to crash, or to be extremely slow. The following changes were made:

- In the `MatchActivity`, the `OnCreate()` method loads the view, and captures the elements of importance in the view, but does not perform synchronously the request sent to the server to obtain the list of users. That way the UI Thread remains active while a new class name `FindMatchAsyncTask` is created under a new package named `Tasks`. Once this task is finished, a callback function found in the `MatchActivity` is invoked. The presence and implementation of this method is enforced by the interface `IFindMatchCompleteCallback`.
- A toast is added when declining the request of a user.
- When looking for match, the match activity will hide the buttons that depend on the results of the async task. When the results of the tasks are received, the buttons re-appear. Texts dependent of the results are also replaced with "Loading..." while waiting for the response. 
- Reverse geocoding now uses `Android.Location` for asynchronicity and compatibility with Android phones.

The following **async methods** are now available from the mapper & gateway:
- `GetUserFacebookProfilePictureAsync()`: This converts the image to a bitmap asynchronously.

Sometimes, async methods are not sufficient, or may require callback methods. It may also be cleaner and more readable to separate the task performed in the background, and abstract that logic from the activities. The following **async tasks** are now available, each with their callback interface:
 - `GetReviewsAverageAsyncTask`: Obtain all the reviews for a given user, and compute the average asynchronously.
- `FindMatchAsyncTask`: Obtains all the matched requests asynchronously.

Note on async tasks: 
Activities have visibility over the Tasks, and the Tasks have visibility over the Mappers.

### CODE REVIEW

**Comment by limphilip** [comment]
Please do NOT close issue #80 yet after merging, we _may_ need to change other methods to be asynchronous in other activities if performance issues arise.

File: mRides-app/Tasks/FindMatchAsyncTask.cs [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+        private IOnFindMatchCompleteCallback callBackObject;
+
+        // List of requests matched
+        List<Request> listOfMatchedRequests;
+
+        // Mapper used to make the request
+        private ConsoleMapper consoleMapper;
+        
+        /// <summary>
+        ///
```
Comment (Ideawin) [inline mRides-app/Tasks/FindMatchAsyncTask.cs]: Don't forget to write a description of each method inside this class along with the parameters :) They are all blank

Comment (limphilip) [reply mRides-app/Tasks/FindMatchAsyncTask.cs]: Good catch! I totally forgot about them. I will add them shortly.

**Comment by Ideawin** [comment]
Good job on this! I just tested it out and the matching activity is loading very fast compared to previously! Merging

**Pull request merged**